### PR TITLE
Update copyright to 2024

### PR DESCRIPTION
**Pull request by MarcelKoch**
Happy new year :fireworks: 

I think the format CI should fail, since it will use the old configuration files without the updated year.

### CODE REVIEW

**Review by yhmtsai — APPROVED** [review]

**Comment by upsj** [comment]
The formatting patch shows a lot more changes than just the copyright year update, maybe we need to rerun it or tune the clang-format config?

**Comment by MarcelKoch** [comment]
I think the extra formatting changes are because we didn't run the `format_header.sh` script on all files? The changes from clang-format are only 3 files, so most of that is actually from the script.

**Review by upsj — APPROVED** [review]

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ repos:
   # The official reuse hook only supports calling lint, so we need our own hook
   - id: reuse-annotate
     name: reuse-annotate
-    entry: reuse annotate --license BSD-3-Clause --copyright 'The Ginkgo authors' --year 2017-2023 --style c
+    entry: reuse annotate --license BSD-3-Clause --copyright 'The Ginkgo authors' --year 2024 --style c --merge-copyright
```
Comment (tcojean) [inline .pre-commit-config.yaml:17]: While we're at it, can we automate the year parameter in a platform portable way? e.g., in python something like
```
from datetime import datetime
datetime.now().strftime('%Y')
```
(maybe with timezone UTC)

Comment (MarcelKoch) [reply .pre-commit-config.yaml:17]: I think this would be something that needed to be changed in reuse. I will see, if I can get a PR there.

**Pull request merged**